src/status_im/contexts/chat/messenger/messages/view.cljs

```diff
@@ -42,6 +42,7 @@
                                 chat-screen-layout-calculations-complete?}]
     (when *screen-loaded?*
       (rn/use-mount #(reset! *screen-loaded?* true)))
+    (rn/use-unmount #(rf/dispatch [:chat/clear-current-chat-id]))
```

Comment:
Hi @ulisesmac,

Thank you very much for working on this and pinging me.

We shouldn’t just remove the current-chat-id from the database; we need to call :chat/close. Could you please check which events are triggered when opening the profile and investigate why the chat is not being closed?

If we are calling pop-to-root, then the chat should have been closed. [pop-to-root](https://github.com/status-im/status-mobile/blob/cc6dcc3636bdb76d0b89ffa1c19841b3de267736/src/status_im/navigation/events.cljs#L69).

Reply:
The chat is closed in terms of UI, but the re-frame data is not cleaned. 

The event `:chat/close` does more things than just cleaning the re-frame value.

When we navigate through a profile link press the chat is not properly closed, I'll check what we do in when we go back so that I can replicate the events when the chat closes.

(The logic is hard to understand since a lot of stateful events are dispatched and I try to avoid creating a new bug).

Reply:
`:chat/close` seems to clear a bunch of cache related stuff and reset some state. So it might be worth calling that if we don't have a way of guaranteeing that `:chat/close` will be called some other way right? For example, if we're leaving the chat view, will we return?

That being said, I wonder how using the `unmount` hook might affect this, for example, would `unmount` be called during any animations or loading screens? If so, then calling `:chat/close` could be problematic.

Thoughts?

Reply:
Yes Sean, I tested it but IDK if there's an unexpected call to `rn/use-unmount` during a specific interaction.

Additionally, I don't know which other mechanism use to properly clean the db when an unexpected navigation happens 🤔

I also tried by removing the condition added in the PR that introduced this bug, but then we should properly solve the "blank screen" issue, seems this is the real solution.

Reply:
I'm using `:chat/close` now